torch/testing/_internal/common_methods_invocations.py

```diff
@@ -88,28 +118,68 @@ class UnaryUfuncInfo(OpInfo):
     def __init__(self,
                  name,  # the string name of the function
                  *,
+                 ref,  # a reference function
                  dtypes=floating_types(),
                  dtypesIfCPU=floating_and_complex_types_and(torch.bfloat16),
                  dtypesIfCUDA=floating_and_complex_types_and(torch.half),
-                 dtypesIfROCM=floating_types_and(torch.half, torch.bfloat16),
+                 dtypesIfROCM=floating_types_and(torch.half),
+                 domain=(None, None),  # the [low, high) domain of the function
+                 handles_large_floats=True,  # whether the op correctly handles large float values (like 1e-20)
+                 handles_extremals=True,  # whether the op correctly handles extremal values (like inf)
+                 handles_complex_extremals=True,  # whether the op correct handles complex extremals (like inf -infj)
                  **kwargs):
         super(UnaryUfuncInfo, self).__init__(name,
                                              dtypes=dtypes,
                                              dtypesIfCPU=dtypesIfCPU,
                                              dtypesIfCUDA=dtypesIfCUDA,
                                              dtypesIfROCM=dtypesIfROCM,
                                              **kwargs)
+        self.ref = ref
+        self.domain = domain
+        self.handles_large_floats = handles_large_floats
+        self.handles_extremals = handles_extremals
+        self.handles_complex_extremals = handles_complex_extremals
 
 L = 20
 M = 10
 S = 5
 
+# Situational skips for tests that fail on particular builds
+_windows_skip = ('.*numerics.*complex.*',) if IS_WINDOWS else tuple()
```

Comment:
OK, now that I actually see the regexes in action, I really really hate them. My basic problem is this: suppose I'm adding a new test to the test suite. Ordinarily, I would expect the name that I pick for this test to not matter (except I am obligated to pick a test name that is unique). However, if I *accidentally* pick a test name that happens to be covered by the existing regex set (and these regexes don't seem to be particularly specific), then I will silently start skipping my tests on these configurations. That seems bad!

Wouldn't it be better to just setup some sort of predicate system, where you do exact match on test names, and then can also say things like "and skip if it's on the complex dtypes"? It seems like you're only really matching one of the tests anyway, here!

Reply:
OK, that sounds good. Let's do test_name with options for device and dtype.

Reply:
Also, I don't think you would even want a regex to hit multiple different tests. Then, the construction of the regex would imply that the bug in question is correlated over the tests. But unless you have actual evidence that this is the case, you probably shouldn't assume this! The bugs might be completely unrelated.

Reply:
I believe this is fixed. We can also refine it in follow-ups.